examples/extended_validation/src/tests.rs

```diff
@@ -0,0 +1,86 @@
+use rocket::testing::MockRequest;
+use rocket::http::Method::*;
+use rocket::http::{ContentType, Status};
+
+use super::rocket;
+
+fn test_login<T>(user: &str, pass: &str, age: &str, status: Status, body: T)
+    where T: Into<Option<&'static str>>
+{
+    let rocket = rocket();
+    let query = &format!("username={}&password={}&age={}", user, pass, age);
+
+    let mut req = MockRequest::new(Post, "/login")
+        .header(ContentType::Form)
+        .body(&query);
+
+    let mut response = req.dispatch_with(&rocket);
+    assert_eq!(response.status(), status);
+    if let Some(string) = body.into() {
+        assert!(response.body().and_then(|body| body.into_string()).map_or(false, |s| s.contains(string)));
```

Comment:
The return value of ```into_string``` is a ```Option<String>``` type. As a result, the above suggestion code may lead to the following compiler error.

```
error: no method named `contains` found for type `std::option::Option<std::string::String>` in the current scope
  --> src/tests.rs:21:73
   |
21 |         assert!(response.body().map_or(false, |body| body.into_string().contains(string)));
   |   
```

That's why the ```and_then().map_or``` chain is needed. The return value of the closure within ```and_then``` can be ```Option::None```. If it does, the ```map_or``` will use the default value, which is false.